samcli/lib/providers/provider.py

```diff
@@ -57,10 +59,13 @@ def __init__(self):
         self.result = {"Function": [], "Layer": []}
 
     def add_function(self, function):
+        if function.inlinecode and function.inlinecode is not None:
+            return
         self.result.get("Function").append(function)
```

Comment:
This is used for much more that build. Why are we adjusting this, vs when we create the list of build only?

Reply:
Changed the place for this check